src/gt4py/next/otf/workflow.py

```diff
@@ -61,7 +71,82 @@ def __call__(self, inp: StartT_contra) -> EndT_co:
 
 
 @dataclasses.dataclass(frozen=True)
-class Step(Generic[StartT, EndT]):
+class NamedStepSequence(Generic[StartT, EndT]):
+    """
+    Workflow with linear succession of named steps.
+
+    Examples:
+    ---------
+    >>> import dataclasses
+
+    >>> def parse(x: str) -> int:
+    ...    return int(x)
+
+    >>> def plus_half(x: int) -> float:
+    ...    return x + 0.5
+
+    >>> def stringify(x: float) -> str:
+    ...    return str(x)
+
+    >>> @dataclasses.dataclass(frozen=True)
+    ... class ParseOpPrint(NamedStepSequence[str, str]):
+    ...    parse: Workflow[str, int]
+    ...    op: Workflow[int, float]
+    ...    print: Workflow[float, str]
+
+    >>> pop = ParseOpPrint(
+    ...    parse=parse,
+    ...    op=plus_half,
+    ...    print=stringify
+    ... )
+
+    >>> pop.step_order
+    ['parse', 'op', 'print']
+
+    >>> pop(73)
+    '73.5'
+
+    >>> def plus_tenth(x: int) -> float:
+    ...   return x + 0.1
+
+
+    >>> pop.replace(op=plus_tenth)(73)
+    '73.1'
+    """
+
+    def __call__(self, inp: StartT) -> EndT:
+        """Compose the steps in the order defined in the `.step_order` class attribute."""
+        step_result: Any = inp
+        for step in [getattr(self, s) for s in self.step_order]:
+            step_result = step(step_result)
```

Comment:
I would prefer something like this so that I can see in the debugger which step is currently processed. Just a suggestion though.
```suggestion
        for step_name in self.step_order:
            step_result = getattr(self, step_name)(step_result)
```

Reply:
should we then add a check beforehand to keep failing fast if the `step_order` does not contain a subset of the available steps?

Reply:
I will change it like this for readability, and defer the fast failing optimization until there is a usecase for it.